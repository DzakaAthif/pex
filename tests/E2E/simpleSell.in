SELL 0 Avocado 50 100;
SELL 1 Avocado 50 101;
SELL 2 Avocado 50 102;
SELL 3 Wine 50 100;
SELL 4 Wine 50 101;
SELL 5 Wine 50 102;