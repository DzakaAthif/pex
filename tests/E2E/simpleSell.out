[PEX] Starting
[PEX] Trading 2 products: Wine Avocado
[PEX] Created FIFO /tmp/pe_exchange_0
[PEX] Created FIFO /tmp/pe_trader_0
[PEX] Starting trader 0 (./trader1)
[PEX] Connected to /tmp/pe_exchange_0
[PEX] Connected to /tmp/pe_trader_0
[PEX] [T0] Parsing command: <SELL 0 Avocado 50 100>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 0
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 1
[PEX]		SELL 50 @ $100 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <SELL 1 Avocado 50 101>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 0
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 2
[PEX]		SELL 50 @ $101 (1 order)
[PEX]		SELL 50 @ $100 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <SELL 2 Avocado 50 102>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 0
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 3
[PEX]		SELL 50 @ $102 (1 order)
[PEX]		SELL 50 @ $101 (1 order)
[PEX]		SELL 50 @ $100 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <SELL 3 Wine 50 100>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 1
[PEX]		SELL 50 @ $100 (1 order)
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 3
[PEX]		SELL 50 @ $102 (1 order)
[PEX]		SELL 50 @ $101 (1 order)
[PEX]		SELL 50 @ $100 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <SELL 4 Wine 50 101>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 2
[PEX]		SELL 50 @ $101 (1 order)
[PEX]		SELL 50 @ $100 (1 order)
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 3
[PEX]		SELL 50 @ $102 (1 order)
[PEX]		SELL 50 @ $101 (1 order)
[PEX]		SELL 50 @ $100 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] [T0] Parsing command: <SELL 5 Wine 50 102>
[PEX]	--ORDERBOOK--
[PEX]	Product: Wine; Buy levels: 0; Sell levels: 3
[PEX]		SELL 50 @ $102 (1 order)
[PEX]		SELL 50 @ $101 (1 order)
[PEX]		SELL 50 @ $100 (1 order)
[PEX]	Product: Avocado; Buy levels: 0; Sell levels: 3
[PEX]		SELL 50 @ $102 (1 order)
[PEX]		SELL 50 @ $101 (1 order)
[PEX]		SELL 50 @ $100 (1 order)
[PEX]	--POSITIONS--
[PEX]	Trader 0: Wine 0 ($0), Avocado 0 ($0)
[PEX] Trader 0 disconnected
[PEX] Trading completed
[PEX] Exchange fees collected: $0
